[scroll-animations] Remove use of members to hold start/endOffset on ViewTimeline

nmoucht opened the pull request:
#### 398ad41dc09198f6b790245ba82686968bdee7b5
<pre>
[scroll-animations] Remove use of members to hold start/endOffset on ViewTimeline
<a href="https://bugs.webkit.org/show_bug.cgi?id=281615">https://bugs.webkit.org/show_bug.cgi?id=281615</a>
<a href="https://rdar.apple.com/138059636">rdar://138059636</a>

Reviewed by Antoine Quint and Tim Nguyen.

Remove members used to hold start/endOffset values and return a Ref instead.

* Source/WebCore/animation/ViewTimeline.cpp:
(WebCore::ViewTimeline::ViewTimeline):
(WebCore::ViewTimeline::startOffset):
(WebCore::ViewTimeline::endOffset):
* Source/WebCore/animation/ViewTimeline.h:

Canonical link: <a href="https://commits.webkit.org/285434@main">https://commits.webkit.org/285434@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5fc8fade1d42ae4ac1805ad988fb9b9c8e8b24e3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/72606 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52031 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25404 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76794 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23819 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/74721 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59836 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23640 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/57138 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15652 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75673 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47080 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62516 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/37565 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43730 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/19983 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22169 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65591 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20342 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78465 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16852 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/19478 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/65590 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16900 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62525 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64859 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/13148 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6797 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11153 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47829 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/2616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48896 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/50191 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/48641 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/animation/ViewTimeline.h:
```diff
@@ -54,8 +54,8 @@ class ViewTimeline final : public ScrollTimeline {
     Element* subject() const { return m_subject.get(); }
-    const CSSNumericValue& startOffset();
-    const CSSNumericValue& endOffset();
+    Ref<CSSNumericValue> startOffset();
+    Ref<CSSNumericValue> endOffset();
```
I wonder if we should call these `computeStartOffset`/`computeEndOffset` since they're not trivial getters.

nmoucht replied:
These are API functions exposed via the IDL. Maybe we could do ImplementedAs to avoid confusion? Although we generally only do that to prefix functions with bindings.

nt1m replied:
I'm fine either way, we could either use `[ImplementedAs]` or leave as is. I wonder if @cdumez or @graouts has opinions.

graouts replied:
Not a big deal to me, this looks OK to me as-is.

nmoucht committed:
[scroll-animations] Remove use of members to hold start/endOffset on ViewTimeline
https://bugs.webkit.org/show_bug.cgi?id=281615
rdar://138059636

Reviewed by Antoine Quint and Tim Nguyen.

Remove members used to hold start/endOffset values and return a Ref instead.

* Source/WebCore/animation/ViewTimeline.cpp:
(WebCore::ViewTimeline::ViewTimeline):
(WebCore::ViewTimeline::startOffset):
(WebCore::ViewTimeline::endOffset):
* Source/WebCore/animation/ViewTimeline.h:

Canonical link: https://commits.webkit.org/285434@main
